backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,382 @@
+            slide.backgroundColor = s.backgroundColor;
+            slide.content = s.content;
+            slide.layout = s.layout;
+            slide.order = s.order;
+            await slide.save();
+        }
+    }
+
+    /**
+     * Generates a random code for a sequence.
+     * @returns  the generated sequence
+     */
+    private static generateCode(): string {
+        const codechar = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+        const charactersLength = codechar.length;
+        let result = '';
+        for (let i = 0; i < length; i++) {
+            result += codechar.charAt(
+                Math.floor(Math.random() * charactersLength)
+            );
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the given user id matches an user in the database
+     * @param userId the id of the user
+     * @returns true if the user exists
+     */
+    private static async isValidUser(userId: number): Promise<boolean> {
+        return (await DBUser.findOneBy({ id: userId })) !== null;
+    }
+    /**
+     * removing the given code from the given list. used on read/write access lists
+     * @param list the list
+     * @param code the code
+     * @returns the list wihtout the code
+     */
+    private static removeFromUserList(list: string[], code: string): string[] {
+        return list.splice(list.indexOf(code), 1);
+    }
+
+    /**
+     * updates users to fullfill a new access list
+     * @param oldAccessList the old access list
+     * @param newAccessList the new access list to fullfill
+     * @param code the code of the sequence
+     */
+    private static async updateSharedList(
+        oldAccessList: number[],
+        newAccessList: number[],
+        code: string
+    ): Promise<void> {
+        for (const uId of newAccessList) {
+            const user = await DBUser.findOneBy({ id: uId });
+            if (user === null) {
+                throw new LoernwerkError(
+                    'A read access user, does not exist',
+                    LoernwerkErrorCodes.NOT_FOUND
+                );
+            }
+            if (!user.sharedSequencesReadAccess.includes(code)) {
+                user.sharedSequencesReadAccess.push(code);
+                await user.save();
+            }
+        }
+        for (const uId of this.getRemovedUser(oldAccessList, newAccessList)) {
+            const user = await DBUser.findOneBy({ id: uId });
+            if (user === null) {
+                continue; //this should be fine, right?
+            }
+            user.sharedSequencesReadAccess = this.removeFromUserList(
+                user.sharedSequencesReadAccess,
+                code
+            );
+            await user.save();
+        }
+    }
+
+    /**
+     * gets the removed user from two access lists
+     * @param oldAccessList the old access list
+     * @param newAccessList the new access list
+     * @returns the removed user (may empty)
+     */
+    private static getRemovedUser(
```

Comment:
Maybe a better name? This sounds like we are getting a user that was removed from the DB